mllib/src/main/scala/org/apache/spark/ml/optim/aggregator/LogisticAggregator.scala

```diff
@@ -209,6 +208,70 @@ private[ml] class LogisticAggregator(
       s"got type ${bcCoefficients.value.getClass}.)")
   }
 
+  // Helper vectors and matrices for binary:
+  @transient private lazy val binaryLinear = {
```

Comment:
So, are these lazy just to deal with recreating them after deserialization? they don't seem big, so can they just be non-transient, non-lazy? unless it's a material problem, might be simpler and faster.
Or how much do you need to hold on to scratch vectors like auxiliaryVec vs just locals?

Reply:
`binaryLinear`, `binaryIntercept`, `multinomialLinear`, `multinomialIntercept` are the linear and bias part of coefficients, repectively.

`binaryLinearGradSumVec` (numFeatures) and `multinomialLinearGradSumMat` (numClassXnumFeatures) are used to store result of `gemv`/`gemm` if `fitIntercept==True`, since `gradientSumArray` contains gradient sums of intercepts and can not be used directly in `gemv`/`gemm`.

`auxiliaryVec` (blockSize) and `multinomialAuxiliaryMat` (blockSizeXnumClasses) are used to store the intermediate multiplication(margins) and multipliers.

they can be used among blocks, and if they are used multi-times in one call we can assign them to local variables.
However I am OK to make them local variables, since I guess they are not the bottleneck.

Reply:
OK up to your judgment. It'd be simpler to not even make them members, if it's not much difference to performance